plugins/inventory/gql_inventory.py

```diff
@@ -189,6 +183,44 @@ def add_variable(self, host: str, var: str, var_type: str):
         """
         self.inventory.set_variable(host, var_type, var)
 
+    def create_groups(self, json_data: dict):
+        """Creates groups for host based on `group_by` parameters."""
+        for device in json_data["data"]["devices"]:
```

Comment:
We should bring the device addition outside of this, but pass in the device_name within `main()`.

Reply:
Updated